### DISCUSSION TITLE
RangeError: Maximum call stack size exceeded instead of timeout

### DISCUSSION THREAD

**Discussion by christianinsert** [cypress-io/cypress#16086] [Questions and Help]
Hi all!
I'm  using Cypress 6.8 for E2E tests. On my CI system, I'm experiencing errors right now when trying to do `cy.request`s:
```
CypressError: `cy.request()` timed out waiting `30000ms` for a response from your server.
The request we sent was:
Method: POST
URL: http://localhost/whatever
No response was received within the timeout.
```
In order to reproduce that issue locally I added in the server route in question a 40s timeout before it returns anything. (I don't think the original problem on the CI is connected to cypress. I just want to be able to debug it locally.) I expected to get the same errors locally than on the CI system. However, what I get is 
```
RangeError: Maximum call stack size exceeded
    at _deconstructPacket (/home/.../.cache/Cypress/6.8.0/Cypress/resources/app/packages/socket/node_modules/socket.io-parser/dist/binary.js:21:28)
```    
and then ca. 15k lines like this.

Now I'm confused.
How come? Is there anything  could do in order to get a `timed out waiting` error instead?

**Comment by bahmutov** [comment]
See https://github.com/cypress-io/cypress/issues/15101 and https://github.com/cypress-io/cypress/issues/15629
